plugin/distribute.go

```diff
@@ -0,0 +1,165 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/bytedance/sonic"
+	"github.com/redis/go-redis/v9"
+	"github.com/vermaShivansh/coraza-ratelimit-plugin/helpers"
+)
+
+type Distributed struct {
+	Active       bool          // if ratelimit has to be distributed
+	SyncInterval time.Duration // syncing with redis at interval
+	lastSync     int64         // timestamp for lastSync with redis
+	UniqueKey    string        // has to be same for all the instances which have to be synced together
+}
+
+var RATELIMIT_KEY = "coraza_ratelimit_key"
+
+// initiate distribution
+func (e *Ratelimit) initDistribute(syncInterval time.Duration) error {
+	var err error
+
+	unique_key := os.Getenv(RATELIMIT_KEY)
+
+	if unique_key == "" {
+		return errors.New("please set a env value with keyname `coraza_ratelimit_key` to use ratelimit distribution ")
+	}
+
+	if err = helpers.CheckRatelimitDistributeKey(unique_key); err != nil {
+		return err
+	}
+
+	e.Distributed.Active = true
+	e.Distributed.UniqueKey = unique_key
+	e.Distributed.SyncInterval = syncInterval
+	e.Distributed.lastSync = 0
+
+	return nil
+}
+
+// it uses syncFunc declared below
+/*
```

Comment:
Please use go comment style `//`

Reply:
Okay. I will convert all multi-line comments to single lines but is there some specific reason for this?
Like is there a specific reason to encourage // comment style in Go ?

Reply:
It is the standard and also it is better to use a single way of commenting.

Reply:
okay thanks for letting me know 😄